### ISSUE TITLE
Cannot read `yarn.lock` larger than 1MB

### ISSUE THREAD

**Issue by vknez** [Simek/yarn-lock-changes#20]
Hi,

Thanks for creating this action - it would be useful for my CI/CD. However, our `yarn.lock` is 1.1MB and the action fails with the following message:

```
Run Simek/yarn-lock-changes@main
  with:
    collapsibleThreshold: 25
    path: yarn.lock
    token: ***
    updateComment: true

Error: This API returns blobs up to 1 MB in size. The requested blob is too large to fetch via the API, but you can use the Git Data API to request blobs up to 100 MB in size.: {"resource":"Blob","field":"data","code":"too_large"}
```

**Comment by Simek** [comment]
Hi @vknez, thank you for testing the action and reporting your problem! 👍 

I will investigate one of possible solutions and will let you know if this have help.

However I would also recommend you to run `npx yarn-deduplicate` (https://github.com/atlassian/yarn-deduplicate) or similar tool, because size of your lock is quite big, even for large monorepo projects standards 😉

**Comment by Simek** [comment]
@vknez Unfortunately I cannot reproduce your issue, so you need to provide me more information or create repro repository.

`yarn.lock` in #21 weights `1.34 MB` and there is no issue with fetching it:
* https://github.com/Simek/yarn-lock-changes/blob/huge-lock/yarn.lock

The only problem was a comment length, which I would like to improve in the future, but for now, the basic fix applied in this PR seems fine.

**Comment by vknez** [comment]
Thanks for looking into this so promptly. 

Are you sure you already had a huge yarn.lock in the repo and then updated it in a commit an the action passed successfully? 

I'll try to make a repro when I get back to the computer, but it seemed to me that the issue was fetching a large file from the GH API.

**Comment by Simek** [comment]
> I'll try to make a repro when I get back to the computer, but it seemed to me that the issue was fetching a large file from the GH API.

Yeah, there is for sure the root of the problem. I had to add fallback to [`getBlob`](https://octokit.github.io/rest.js/v18#git-get-blob), but to make sure it did not break the action for other users I had to perform series of tests.

 I have one more idea how to repro this problem outside this repo, but it will make way harder to test and debug it, but I will try that

**Comment by Simek** [comment]
@vknez I think that I was able to fix the issue. Please retest the action in your repository using `main`, if it will be successful I will release new version to marketplace with a fix included, if it will fail, feel free to reopen this issue with the failure details. 🙂

**Comment by vknez** [comment]
Thanks @Simek! It works perfectly now.

Speaking of large `yarn.lock`, I did try https://github.com/atlassian/yarn-deduplicate, but it created a lot of issues because of different minor versions of npm packages needed by various local monorepo packages.
